utils/endOfLifeDateApi/fetchAllDependencyEolInfo.js

```diff
@@ -0,0 +1,73 @@
+import { EndOfLifeDateApiClient } from "./index.js";
+import { readFromJsonFile } from "../index.js"
+import { writeFile, mkdir } from "fs/promises";
+import path from "path";
+
+/**
+ * @typedef {Object} DependencyLifetimes
+ * @property {string} dependency - The name of the dependency
+ * @property {import("./index.js").Cycle[]} lifetimes - The lifetimes of each release cycle for this dependency
+ */
+
+/**
+ * Fetches all lifetime information for all repository dependencies from the endoflife.date
+ * API.
+ * Dependencies for which no information could be found are ignored.
+ * @returns {Promise<DependencyLifetimes[]>}
+ */
+const fetchAllDependencyLifetimes = async () => {
+  const pathToRepos = "./data/repos.json";
+  const persistedRepoData = await readFromJsonFile(pathToRepos);
+
+  const dependencySet = new Set();
+  persistedRepoData.repos.map((repo) => repo.dependencies).forEach((dependencies) => {
```

Comment:
up to you but I feel like this could be a good use case for [flatMap](https://devdocs.io/javascript/global_objects/array/flatmap)

Reply:
I think I just forgot that `flatMap` existed - I've made the change and I agree it looks cleaner.

Reply:
Yeah fair enough, it's rare that I see a use case for it but here's one